### PR TITLE
Fedora changes:

### PR DESCRIPTION
**Pull request by maxamillion**
- ansible bootstrap playbook for Fedora 23+
  - add conditionals to handle yum vs dnf
  - add Fedora OpenShift COPR
  - update BYO host README for repo configs and fedora bootstrap

NOTE:
OpenShift doesn't currently work because of some bugs in Fedora 23 land[[0](https://bugzilla.redhat.com/show_bug.cgi?id=1276170)][[1](https://bugzilla.redhat.com/show_bug.cgi?id=1234757)][[2](https://bugzilla.redhat.com/show_bug.cgi?id=1285039)] - thanks to eparis for the help -, but this pull request does allow a successful origin deployment on Fedora. Also note that I plan to remove the special case [Fedora COPR for Openshift](https://copr.fedoraproject.org/coprs/maxamillion/fedora-openshift/) once OpenShift is successfully brought into the Fedora official repositories via the [Fedora OpenShift Change Proposal](https://fedoraproject.org/wiki/Changes/OpenShiftOrigin). I will open another pull request to fix that up when the time comes.

### CODE REVIEW

File: roles/etcd/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ Requirements
 ------------
 
 This role assumes it's being deployed on a RHEL/Fedora based host with package
-named 'etcd' available via yum.
+named 'etcd' available via yum for dnf (conditionally).
```
Comment (detiber) [inline roles/etcd/README.md]: nitpicky, but s/for/or/

File: roles/openshift_node/tasks/main.yml [hunk 2]
```diff
@@ -39,12 +39,23 @@
...
 - name: Install sdn-ovs package
   yum: pkg={{ openshift.common.service_type }}-sdn-ovs{{ openshift_version }} state=present
   register: sdn_install_result
-  when: openshift.common.use_openshift_sdn
+  when: (ansible_pkg_mgr == "yum") and openshift.common.use_openshift_sdn
+
+- name: Install sdn-ovs package
+  dnf: pkg={{ openshift.common.service_type }}-sdn-ovs{{ openshift_version }} state=present
+  register: sdn_install_result
+  when: (ansible_pkg_mgr == "dnf") and openshift.common.use_openshift_sdn
```
Comment (detiber) [inline roles/openshift_node/tasks/main.yml]: strictly speaking the parens are not needed here, but I have no objection to keeping them.

**Comment by detiber** [comment]
@maxamillion thanks for submitting this! outside of my mostly nitpicky comments, :+1: from me.

On a side note, I can't wait until we can declare ansible 2.0 a requirement, because being able to just call the package task instead of conditionalizing all of the package installs would be nice.

@sdodson heads up this has potential to impact your containerization PR.

**Comment by sdodson** [comment]
Thanks for the heads up. Other than being surprised that this is necessary it all looks good.

LGTM

**Comment by maxamillion** [comment]
@detiber Thanks for the review. I'll fix up the things you mentioned. Also massive +1 for Ansible 2.0 ... super excited about it.

**Comment by maxamillion** [comment]
@sdodson - Unfortunately yes, at least for now. In Ansible 2.0 this won't be necessary (it'll still work, just won't be necessary) because they are adding a generic `package` module. Certain aspects of using the Ansible `yum` module on a Fedora 23+ system error because of missing yum python libraries due to the `dnf` transition.

**Comment by detiber** [comment]
@twiest ptal

**Comment by detiber** [comment]
@twiest @wshearn bump for review on this

**Comment by maxamillion** [comment]
I rebased on master, fixed merge conflicts and fixed the repo handler to work with both yum and dnf

**Comment by maxamillion** [comment]
Crap ... I need to fix some things, the merge went bad

**Comment by detiber** [comment]
@maxamillion thanks again for working on this!

**Comment by maxamillion** [comment]
Alright, rebased, fixed merge conflicts and squashed commits.

**Comment by detiber** [comment]
@twiest @wshearn ptal I'd like to get this merged before any further merge conflicts crop up.

File: roles/openshift_repos/handlers/main.yml [hunk 3]
```diff
@@ -1,3 +1,6 @@
 ---
-- name: refresh package cache
+- name: refresh yum cache
+  command: yum clean all
+
+- name: refresh dnf cache
   command: yum clean all
```
Comment (wshearn) [inline roles/openshift_repos/handlers/main.yml]: Shouldn't this be `dnf clean all`?

**Comment by wshearn** [comment]
After that we get clarification or change based on my note it is a :+1: from me.

Though eventually I would like to see a package install role where we can have the `when ansible_pkg_mgr == "{dnf,yum}"` bits to cut down on dupe code.

**Comment by detiber** [comment]
@wshearn consider this temporary until we can require ansible 2.0, then we can use the new and improved package module.

**Comment by wshearn** [comment]
ok, :+1: from me.

**Comment by detiber** [comment]
@wshearn if you are good with this, can you merge?

**Pull request merged**